File: Leetcode_重刷紀錄/0016.Problems1004. Max Consecutive Ones III/MaxConsecutiveOnesIII.cpp
//1004. Max Consecutive Ones III
/* 這邊是實作sliding window 的方法，實現方式類似雙指針，不過主要裝的是同一區域的數值
*/
class Solution {
public:
    int longestOnes(vector<int>& nums, int k) {
        int left = 0, right = 0;
        int max_length = 0;
        int zero_count = 0;

        while (right < nums.size()) {
            if (nums[right] == 0) {
                zero_count++;
            }

            while (zero_count > k) {
                if (nums[left] == 0) {
                    zero_count--;
                }
                left++;
            }

            max_length = max(max_length, right - left + 1);
            right++;
        }

        return max_length;
    }
};

File: Leetcode_重刷紀錄/0016.Problems1004. Max Consecutive Ones III/MaxConsecutiveOnesIII.py
#1004. Max Consecutive Ones III
# 這邊是實作sliding window 的方法，實現方式類似雙指針，不過主要裝的是同一區域的數值

class Solution:
    def longestOnes(self, nums: List[int], k: int) -> int:
        l=r=0    
        for r in range(len(nums)):
            if nums[r] == 0:
                k-=1
            if k<0:
                if nums[l] == 0:
                    k+=1
                l+=1
        return r-l+1